// @aws-sdk/client-comprehendmedical/dist-es/commands/DescribeICD10CMInferenceJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeICD10CMInferenceJobRequest, DescribeICD10CMInferenceJobResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeICD10CMInferenceJobCommand, serializeAws_json1_1DescribeICD10CMInferenceJobCommand, } from "../protocols/Aws_json1_1";
var DescribeICD10CMInferenceJobCommand = (function (_super) {
    __extends(DescribeICD10CMInferenceJobCommand, _super);
    function DescribeICD10CMInferenceJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeICD10CMInferenceJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "DescribeICD10CMInferenceJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeICD10CMInferenceJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeICD10CMInferenceJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeICD10CMInferenceJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeICD10CMInferenceJobCommand(input, context);
    };
    DescribeICD10CMInferenceJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeICD10CMInferenceJobCommand(output, context);
    };
    return DescribeICD10CMInferenceJobCommand;
}($Command));
export { DescribeICD10CMInferenceJobCommand };
