// @aws-sdk/client-comprehendmedical/dist-es/commands/ListICD10CMInferenceJobsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListICD10CMInferenceJobsRequest, ListICD10CMInferenceJobsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListICD10CMInferenceJobsCommand, serializeAws_json1_1ListICD10CMInferenceJobsCommand, } from "../protocols/Aws_json1_1";
var ListICD10CMInferenceJobsCommand = (function (_super) {
    __extends(ListICD10CMInferenceJobsCommand, _super);
    function ListICD10CMInferenceJobsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListICD10CMInferenceJobsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "ListICD10CMInferenceJobsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListICD10CMInferenceJobsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListICD10CMInferenceJobsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListICD10CMInferenceJobsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListICD10CMInferenceJobsCommand(input, context);
    };
    ListICD10CMInferenceJobsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListICD10CMInferenceJobsCommand(output, context);
    };
    return ListICD10CMInferenceJobsCommand;
}($Command));
export { ListICD10CMInferenceJobsCommand };
