// @aws-sdk/client-comprehendmedical/dist-es/commands/ListRxNormInferenceJobsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListRxNormInferenceJobsRequest, ListRxNormInferenceJobsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListRxNormInferenceJobsCommand, serializeAws_json1_1ListRxNormInferenceJobsCommand, } from "../protocols/Aws_json1_1";
var ListRxNormInferenceJobsCommand = (function (_super) {
    __extends(ListRxNormInferenceJobsCommand, _super);
    function ListRxNormInferenceJobsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListRxNormInferenceJobsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "ListRxNormInferenceJobsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListRxNormInferenceJobsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListRxNormInferenceJobsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListRxNormInferenceJobsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListRxNormInferenceJobsCommand(input, context);
    };
    ListRxNormInferenceJobsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListRxNormInferenceJobsCommand(output, context);
    };
    return ListRxNormInferenceJobsCommand;
}($Command));
export { ListRxNormInferenceJobsCommand };
