// @aws-sdk/client-comprehendmedical/dist-es/commands/ListSNOMEDCTInferenceJobsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSNOMEDCTInferenceJobsRequest, ListSNOMEDCTInferenceJobsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListSNOMEDCTInferenceJobsCommand, serializeAws_json1_1ListSNOMEDCTInferenceJobsCommand, } from "../protocols/Aws_json1_1";
var ListSNOMEDCTInferenceJobsCommand = (function (_super) {
    __extends(ListSNOMEDCTInferenceJobsCommand, _super);
    function ListSNOMEDCTInferenceJobsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSNOMEDCTInferenceJobsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "ListSNOMEDCTInferenceJobsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSNOMEDCTInferenceJobsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListSNOMEDCTInferenceJobsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSNOMEDCTInferenceJobsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListSNOMEDCTInferenceJobsCommand(input, context);
    };
    ListSNOMEDCTInferenceJobsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListSNOMEDCTInferenceJobsCommand(output, context);
    };
    return ListSNOMEDCTInferenceJobsCommand;
}($Command));
export { ListSNOMEDCTInferenceJobsCommand };
