// @aws-sdk/client-comprehendmedical/dist-es/commands/StartPHIDetectionJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartPHIDetectionJobRequest, StartPHIDetectionJobResponse } from "../models/models_0";
import { deserializeAws_json1_1StartPHIDetectionJobCommand, serializeAws_json1_1StartPHIDetectionJobCommand, } from "../protocols/Aws_json1_1";
var StartPHIDetectionJobCommand = (function (_super) {
    __extends(StartPHIDetectionJobCommand, _super);
    function StartPHIDetectionJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartPHIDetectionJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "StartPHIDetectionJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartPHIDetectionJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartPHIDetectionJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartPHIDetectionJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartPHIDetectionJobCommand(input, context);
    };
    StartPHIDetectionJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartPHIDetectionJobCommand(output, context);
    };
    return StartPHIDetectionJobCommand;
}($Command));
export { StartPHIDetectionJobCommand };
