// @aws-sdk/client-comprehendmedical/dist-es/commands/StartEntitiesDetectionV2JobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartEntitiesDetectionV2JobRequest, StartEntitiesDetectionV2JobResponse } from "../models/models_0";
import { deserializeAws_json1_1StartEntitiesDetectionV2JobCommand, serializeAws_json1_1StartEntitiesDetectionV2JobCommand, } from "../protocols/Aws_json1_1";
var StartEntitiesDetectionV2JobCommand = (function (_super) {
    __extends(StartEntitiesDetectionV2JobCommand, _super);
    function StartEntitiesDetectionV2JobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartEntitiesDetectionV2JobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "StartEntitiesDetectionV2JobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartEntitiesDetectionV2JobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartEntitiesDetectionV2JobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartEntitiesDetectionV2JobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartEntitiesDetectionV2JobCommand(input, context);
    };
    StartEntitiesDetectionV2JobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartEntitiesDetectionV2JobCommand(output, context);
    };
    return StartEntitiesDetectionV2JobCommand;
}($Command));
export { StartEntitiesDetectionV2JobCommand };
