// @aws-sdk/client-comprehendmedical/dist-es/commands/StartRxNormInferenceJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartRxNormInferenceJobRequest, StartRxNormInferenceJobResponse } from "../models/models_0";
import { deserializeAws_json1_1StartRxNormInferenceJobCommand, serializeAws_json1_1StartRxNormInferenceJobCommand, } from "../protocols/Aws_json1_1";
var StartRxNormInferenceJobCommand = (function (_super) {
    __extends(StartRxNormInferenceJobCommand, _super);
    function StartRxNormInferenceJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartRxNormInferenceJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "StartRxNormInferenceJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartRxNormInferenceJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartRxNormInferenceJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartRxNormInferenceJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartRxNormInferenceJobCommand(input, context);
    };
    StartRxNormInferenceJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartRxNormInferenceJobCommand(output, context);
    };
    return StartRxNormInferenceJobCommand;
}($Command));
export { StartRxNormInferenceJobCommand };
