// @aws-sdk/client-comprehendmedical/dist-es/commands/StopSNOMEDCTInferenceJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopSNOMEDCTInferenceJobRequest, StopSNOMEDCTInferenceJobResponse } from "../models/models_0";
import { deserializeAws_json1_1StopSNOMEDCTInferenceJobCommand, serializeAws_json1_1StopSNOMEDCTInferenceJobCommand, } from "../protocols/Aws_json1_1";
var StopSNOMEDCTInferenceJobCommand = (function (_super) {
    __extends(StopSNOMEDCTInferenceJobCommand, _super);
    function StopSNOMEDCTInferenceJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopSNOMEDCTInferenceJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "StopSNOMEDCTInferenceJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopSNOMEDCTInferenceJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopSNOMEDCTInferenceJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopSNOMEDCTInferenceJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StopSNOMEDCTInferenceJobCommand(input, context);
    };
    StopSNOMEDCTInferenceJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StopSNOMEDCTInferenceJobCommand(output, context);
    };
    return StopSNOMEDCTInferenceJobCommand;
}($Command));
export { StopSNOMEDCTInferenceJobCommand };
