// @aws-sdk/client-comprehendmedical/dist-es/commands/StopEntitiesDetectionV2JobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopEntitiesDetectionV2JobRequest, StopEntitiesDetectionV2JobResponse } from "../models/models_0";
import { deserializeAws_json1_1StopEntitiesDetectionV2JobCommand, serializeAws_json1_1StopEntitiesDetectionV2JobCommand, } from "../protocols/Aws_json1_1";
var StopEntitiesDetectionV2JobCommand = (function (_super) {
    __extends(StopEntitiesDetectionV2JobCommand, _super);
    function StopEntitiesDetectionV2JobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopEntitiesDetectionV2JobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendMedicalClient";
        var commandName = "StopEntitiesDetectionV2JobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopEntitiesDetectionV2JobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopEntitiesDetectionV2JobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopEntitiesDetectionV2JobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StopEntitiesDetectionV2JobCommand(input, context);
    };
    StopEntitiesDetectionV2JobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StopEntitiesDetectionV2JobCommand(output, context);
    };
    return StopEntitiesDetectionV2JobCommand;
}($Command));
export { StopEntitiesDetectionV2JobCommand };
